libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/PlannedRouteRefreshController.kt

```diff
@@ -138,13 +146,17 @@ internal class PlannedRouteRefreshController @VisibleForTesting constructor(
                 if (it.anySuccess()) {
                     stateHolder.onSuccess()
                     listener.onRoutesRefreshed(it)
+                    val refreshedRoutes = listOf(it.primaryRouteRefresherResult.route) +
+                        it.alternativesRouteRefresherResults.map { it.route }
+                    routesToRefresh = refreshedRoutes
+                    scheduleNewUpdate(refreshedRoutes, false)
```

Comment:
I really like that route refresh controller doesn't wait for new routes set from directions session to start the next cycle of refresh. That decreases coupling + this dependency has never been obvious to me.

Reply:
++, it took me a while to understand how that works.